in_toto/process.py

```diff
@@ -0,0 +1,124 @@
+            respectively. Valid values are PIPE, DEVNULL, an existing file
+            descriptor (a positive integer), an existing file object, and None.
+            PIPE indicates that a new pipe to the child should be created.
+            DEVNULL indicates that the special file os.devnull will be used.
+            With the default settings of None, no redirection will occur; the
+            child's file handles will be inherited from the parent.
+            Additionally, stderr can be STDOUT, which indicates that the stderr
+            data from the applications should be captured into the same file
+            handle as for stdout."
+
+    timeout:
+            "The timeout argument is passed to Popen.communicate(). If the
+            timeout expires, the child process will be killed and waited for.
+            The TimeoutExpired exception will be re-raised after the child
+            process has terminated."
+
+    universal_newlines:
+            "If universal_newlines is False the file objects stdin, stdout and
+            stderr will be opened as binary streams, and no line ending
+            conversion is done."
+
+            "If universal_newlines is True, these file objects will be opened
+            as text streams in universal newlines mode using the encoding
+            returned by locale.getpreferredencoding(False). For stdin, line
+            ending characters '\n' in the input will be converted to the
+            default line separator os.linesep. For stdout and stderr, all line
+            endings in the output will be converted to '\n'. For more
+            information see the documentation of the io.TextIOWrapper class
+            when the newline argument to its constructor is None."
+
+  <Exceptions>
+    securesystemslib.exceptions.FormatError:
+            If the cmd as list does not match
+            in_toto.formats.LIST_OF_ANY_STRING_SCHEMA.
+
+    OSError:
+            If the given command is not present or non-executable.
+
+    subprocess.TimeoutExpired:
+            If the process does not terminate after timeout seconds.
+
+  <Side Effects>
+    The side effects of executing the given command in this environment.
+
+  <Returns>
+    A subprocess.CompletedProcess instance.
+
+  """
+  if isinstance(cmd, six.string_types):
+    cmd = shlex.split(cmd)
+  else:
+    formats.LIST_OF_ANY_STRING_SCHEMA.check_match(cmd)
+
+  # The reason why we are not allowed to even specify stdin=None when input
+  # is specified is due to this overly stringent code in subprocess:
+  # https://github.com/google/python-subprocess32/blob/560f1a92db18c2d2bebe4049756528ce827aa366/subprocess32.py#L402
+  if _input:
+    log.debug('Ignoring stdin: '+str(stdin))
+    return subprocess.run(cmd, check=check, input=_input,
+      stdout=stdout, stderr=stderr, timeout=timeout,
+      universal_newlines=universal_newlines)
+  else:
+    log.debug('Ignoring input: '+str(input))
```

Comment:
This is not a proper review yet, I just noticed something while skimming through your PR:
Above `input` should be an `_input`, right?

As for Python3.4 compatibility, why not use `check_output`, as you suggested in the corresponding issue (#219)? Let me know if you want me to take a more thorough look at it.